File: code_samples/swift/functional_higher_order.swift
import Foundation

/// Functional: Higher Order
/// AI/ML Training Sample

class HigherOrder {
    var data: String
    
    init() {
        self.data = ""
    }
    
    func process(_ input: String) {
        self.data = input
    }
    
    func getData() -> String {
        return self.data
    }
    
    func validate() -> Bool {
        return !self.data.isEmpty
    }
}

// Example usage
let instance = HigherOrder()
instance.process("example")
print("Data: \(instance.getData())")
print("Valid: \(instance.validate())")

File: code_samples/java/FunctionalHigherOrder.java
/**
 * Functional: Higher Order
 * AI/ML Training Sample
 */
public class FunctionalHigherOrder {
    
    private String data;
    
    public FunctionalHigherOrder() {
        this.data = "";
    }
    
    public void process(String input) {
        this.data = input;
    }
    
    public String getData() {
        return this.data;
    }
    
    public boolean validate() {
        return this.data != null && !this.data.isEmpty();
    }
    
    public static void main(String[] args) {
        FunctionalHigherOrder instance = new FunctionalHigherOrder();
        instance.process("example");
        System.out.println("Data: " + instance.getData());
        System.out.println("Valid: " + instance.validate());
    }
}
